source/components/Timeago/index.js

```diff
@@ -1,70 +1,81 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-/* istanbul ignore next */
-function getTimeDistanceString(datetime) {
-    const date = new Date(parseInt(datetime, 10));
-    const now = Date.now();
-    const diffInSeconds = (now - date) / 1000;
+export const SECOND_MILLISECONDS = 1000;
+export const MINUTE_SECONDS = 60;
+export const HOUR_SECONDS = 60 * MINUTE_SECONDS;
+export const DAY_SECONDS = 24 * HOUR_SECONDS;
+export const FIVE_DAYS_SECONDS = 5 * DAY_SECONDS;
 
-    if (diffInSeconds > 432000) {
-        // more than 5 days ago - show date
-        return date.toLocaleDateString();
-    }
-
-    if (diffInSeconds > 86400) {
-        // more than a day ago
-        return `${Math.round(diffInSeconds / 60 / 60 / 24)}d`;
-    }
-
-    if (diffInSeconds > 3600) {
-        // more than an hour ago
-        return `${Math.round(diffInSeconds / 60 / 60)}h`;
-    }
-
-    if (diffInSeconds < 60) {
-        // less than a minute ago
-        return 'now';
-    }
-
-    return `${Math.round(diffInSeconds / 60)}m`;
-}
+export const TIMEAGO_SHOW_DATE_THRESHOLD = FIVE_DAYS_SECONDS;
+export const TIMEAGO_NOW_THRESHOLD = MINUTE_SECONDS;
 
 /**
  * The Timeago component is a small component that
  * shows the number of seconds/minutes/days from given datetime.
- *
- * It all happens after the component is mounted so it's safe to use this
- * component on the Back-End without messing up the hydration.
  */
-export default class Timeago extends React.Component {
-    state = {
-        display: this.props.datetime,
-    };
+export default function Timeago({
+    datetime, renderNow, renderDate, renderDays, renderHours, renderMinutes,
```

Comment:
Can `render*` function return JSX/JS string? Ideally Id like to use `renderNow={() => <strong>NOW!</string>}`

Reply:
You can return anything because it will just call your `render*` function and return its result. Check this fragment out: `return typeof (renderNow) === 'function' ? renderNow() : 'now';` So sure, you can use it like you did it above :)